src/lib/crypto/kimchi_bindings/stubs/src/srs.rs

```diff
@@ -83,11 +83,14 @@ macro_rules! impl_srs {
                         .unwrap()
                 })?;
 
-                let evals = (0..domain_size)
-                    .map(|j| if i == j { <$F as ark_ff::One>::one() } else { <$F as ark_ff::Zero>::zero() })
-                    .collect();
-                let p = Evaluations::<$F>::from_vec_and_domain(evals, x_domain).interpolate();
-                Ok(srs.commit_non_hiding(&p, None).into())
+                {
+                    // We're single-threaded, so it's safe to grab this pointer as mutable.
+                    // Do not try this at home.
```

Comment:
I'm at home right now!  Should I be worried? ;-)

Reply:
Don't worry, I tried it at home instead